### ISSUE TITLE
Allow reusing django's decorators

### ISSUE THREAD

**Issue by baseplate-admin** [vitalik/django-ninja#775]
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. Ex. I'm always frustrated when `django-ninja` doesn't allow us to use django's decorators as a way to achieve functionality.

Lets say for example i want my users to be logged in on a specific route.

I can not use `from django.contrib.auth.decorators import login_required`.

**Describe the solution you'd like**
I want to reuse the functionality provided by `django` instead of fighting with it.

** Refs :**
* [Vitalik originally wanted work on this](https://github.com/vitalik/django-ninja/issues/725#issuecomment-1505449961).

**Comment by ankushagar99** [comment]
the problem with this that this works only on when you render html template in django but apis return data not render a html page. so as it's return check login and then return redirect url which is based on django templates.

**Comment by igaul** [comment]
Per handler auth is already supported in the decorator and like @ankushagar99  pointed out, django's login_required would not work correctly with rest endpoints.

**Comment by baseplate-admin** [comment]
Fair enough.

My use case is to use django's permissions with `django-ninja`. ( Specifically the [`user_passes_test`](https://docs.djangoproject.com/en/4.2/topics/auth/default/#django.contrib.auth.decorators.user_passes_test) ) 

This doesn't work because ninja adds user to `request.auth`  instead of `request.user`

**Comment by igaul** [comment]
I don't think that would work intuitively with an api anyways, since instead of redirect, you'd want to return a 403.

A ninja provided decorator might be nice for this, but its only a few lines to make for your project.

**Comment by vitalik** [comment]
@igaul 

the next (1.0) version will allow to use django views decorators like this:

```Python
from django.views.decorators.cache import cache_page
from ninja.decorators import decorate_view

@api.get("/some")
@decorate_view(cache_page(60 * 15))  # <---- !!!
def some_operation(request):
     return {"hello": "World"}


```

**Comment by igaul** [comment]
@vitalik 

A decorator to apply decorators :)

Should probably emphasize that this won't work for @baseplate-admin , things like login_required and user_passes_test won't work and you'll probably get more issues opened when people try.

Better solution would probably be exposing just ones that work, like cache settings. And adding a passes_auth_test or similar, though that can already be accomplished via auth.

**Comment by vitalik** [comment]
@igaul

> @vitalik
> 
> A decorator to apply decorators :)

yeah, not the most beautiful solution, but it works.. and at least gives some hooks to make it nicer with router inhertance if some need to use ti more extensively and nicer:


```Python


class RouterWithCache(Router):
    def get(self, path: str, *, cache=None, **kwargs):
           decorator = super().get(path, **kwargs)
           if cache:
                 return decorate_view(cache_page(cache))(decorator)
          return decorator

# and then use like:


@router.get('/some', cache=60*5)
def some_view_that_better_be_cached(request):
     ...          


```

**Comment by igaul** [comment]
Nice, I hadn't considered extending Router, that would actually be a good solution for @baseplate-admin to add user checks without monkey patching django fns.
